### PR TITLE
feat(GiniHealthSDK): Use payment_state instead of checkIfDocumentIsPayable

### PR DESCRIPTION
**Pull request by razvancapra**
https://ginis.atlassian.net/browse/IPC-296

IPC-296
+ fix tests

### CODE REVIEW

File: HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift [hunk 1]
```diff
@@ -20,7 +25,11 @@ import Foundation
     
     /// The line item compound extractions.
     public var lineItems: [[Extraction]]?
-        
+
+    public var isPayable: Bool {
+        extractions.first(where: {$0.name == "payment_state"})?.value == PaymentState.payable.rawValue
```
Comment (motokotomax) [inline HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift:30]: Proposition: Maybe we can define enum with extraction names?

File: HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/AppCoordinator.swift [hunk 2]
```diff
@@ -186,35 +185,34 @@ final class AppCoordinator: Coordinator {
...
+                                            Log("Successfully fetched extractions for id: \(compositeDocument.id)", event: .success)
                                             let invoice = DocumentWithExtractions(documentID: compositeDocument.id,
-                                                                                  extractions: extractions,
-                                                                                  isPayable: isPayable)
+                                                                                  extractionResult: extractionResult)
                                             self?.showInvoicesList(invoices: [invoice])
-                                        case .failure(let error):
-                                            print("❌ Checking if document is payable failed: \(String(describing: error))")
+                                        case let .failure(error):
+                                            Log("Obtaining extractions from document with id \(compositeDocument.id) failed with error: \(String(describing: error))", event: .error)
```
Comment (motokotomax) [inline HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/AppCoordinator.swift:205]: should we use Log here? old code use print

File: HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift (hunk 1 above)
Comment (razvancapra) [reply HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift]: Thanks for the proposition. Adopted!

**Review by zladzeyka — CHANGES_REQUESTED** [review]
Great job @razvancapra 
We need to keep the public interface especially method `healthSDK.checkIfDocumentIsPayable` and change the functionality inside this method, otherwise it's a breaking change 🙈

**Comment by razvancapra** [comment]
> Great job @razvancapra We need to keep the public interface especially method `healthSDK.checkIfDocumentIsPayable` and change the functionality inside this method, otherwise it's a breaking change 🙈

Reverted and updated!

File: HealthSDK/GiniHealthSDK/Tests/GiniHealthSDKTests/GiniHealthTests.swift [hunk 3]
```diff
@@ -55,6 +55,34 @@ final class GiniHealthTests: XCTestCase {
         XCTAssertEqual(receivedProviders, expectedProviders)
     }
 
+    func testDcumentIsPayable() {
```
Comment (zladzeyka) [inline HealthSDK/GiniHealthSDK/Tests/GiniHealthSDKTests/GiniHealthTests.swift:58]:
```suggestion
    func testDocumentIsPayable() {
```

File: HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift [hunk 4]
```diff
@@ -20,7 +39,11 @@ import Foundation
     
     /// The line item compound extractions.
     public var lineItems: [[Extraction]]?
-        
+
+    public var isPayable: Bool {
```
Comment (zladzeyka) [inline HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift:43]: @razvancapra what is the benefit of having 2 interface points that are doing the same thing?
why do we need this variable?

**Review by zladzeyka — CHANGES_REQUESTED** [review]
@razvancapra thank you 🙏
Please find a tiny comment from my side.

File: HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift (hunk 4 above)
Comment (razvancapra) [reply HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift:43]: I've added this one so the client doesn't need to handle parsing on this item. But I can remove it and leave it totally on their side.

**Review by zladzeyka — APPROVED** [review]

**Pull request merged**